operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java

```diff
@@ -117,6 +117,26 @@ public static Labels fromMap(Map<String, String> labels) {
         return new Labels(labels);
     }
 
+    /**
+     * Parse Labels from String into Labels object. The expected format of the String with labels is `key1=value1,key2=value2`
+     *
+     * @param stringLabels  String with labels
+     * @return  Labels object with parsed labels
+     */
+    public static Labels fromString(String stringLabels) {
+        Map<String, String> labels = new HashMap<>();
+
+        if (stringLabels != null && !stringLabels.isEmpty()) {
+            String[] labelsArray = stringLabels.split(",");
+            for (String label : labelsArray) {
+                String[] fields = label.split("=");
+                labels.put(fields[0].trim(), fields[1].trim());
```

Comment:
Rather than throwing AIOOBE this method should throw IAE if the string is not valid according to the syntax it supports. Why? AIOOBE is impl specific; it's a much better contract to have a single exception type to cover the illegal argument case